src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/MinLengthAttribute.cs

```diff
@@ -60,11 +63,16 @@ public override bool IsValid(object value)
             {
                 length = str.Length;
             }
+            else if (value is ICollection collection)
```

Comment:
Factor out code for getting the Count/Length into a helper to DRY it up.

Reply:
I actually started it like this, but then thought let's keep the classes with less methods.
The `MinLengthAttribute` and `MaxLengthAttribute` have a lot of shared code. That's why I kept the implementation separate too.